src/main/resources/META-INF/rewrite/testcontainers.yml

```diff
@@ -0,0 +1,118 @@
+#
+# Copyright 2023 the original author or authors.
+# <p>
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# <p>
+# https://www.apache.org/licenses/LICENSE-2.0
+# <p>
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+---
+type: specs.openrewrite.org/v1beta/recipe
+name: org.openrewrite.java.testing.testcontainers.TestContainersBestPractices
+displayName: Testcontainers best practices
+description: Apply best practices to Testcontainers usage.
+recipeList:
+  # Upgrade to latest version of Testcontainers
+  - org.openrewrite.java.dependencies.UpgradeDependencyVersion:
+      groupId: org.testcontainers
+      artifactId: "*"
+      newVersion: 1.19.x
+  - org.openrewrite.java.testing.testcontainers.ExplicitContainerImages
+  - org.openrewrite.java.testing.testcontainers.GetHostMigration
+---
+type: specs.openrewrite.org/v1beta/recipe
+name: org.openrewrite.java.testing.testcontainers.GetHostMigration
+displayName: Replace `ContainerState.getContainerIpAddress()` with `getHost()`
+description: Replace `org.testcontainers.containers.ContainerState.getContainerIpAddress()` with `getHost()`.
+recipeList:
+  - org.openrewrite.java.ChangeMethodName:
+      methodPattern: org.testcontainers.containers.ContainerState getContainerIpAddress()
+      newMethodName: getHost
+---
+type: specs.openrewrite.org/v1beta/recipe
+name: org.openrewrite.java.testing.testcontainers.ExplicitContainerImages
+displayName: Explicit container images and versions
+description: Replace implicit default container images and versions with explicit versions.
+recipeList:
```

Comment:
kafka, localstack, mockserver and pulsar should be excluded due to the `constructor(String)` uses the tag instead of the image. For those cases, it should use `DockerImageName.parse(<image>:<tag>)`

Reply:
Good to know! I've excluded these for now; likely need an explicit recipe that adds that `DockerImageName.parse`, which we can add separately after this first merge; should be relatively straightforward now there's no build issues.

Reply:
Added an option and restored those four in https://github.com/openrewrite/rewrite-testing-frameworks/pull/433/commits/0ac879ac093c545c8677e74e7499fb033bd1312b